java/ql/src/semmle/code/java/Generics.qll

```diff
@@ -194,7 +194,18 @@ class TypeVariable extends BoundedType, @typevariable {
  * and the second wildcard has a lower bound of `Float`.
  */
 class Wildcard extends BoundedType, @wildcard {
-  /** Holds if this wildcard has an upper bound. */
+  /**
+   * Holds if this wildcard is either unconstrained (i.e. `?`) or
+   * has a type bound.
+   */
+  override predicate hasTypeBound() {
+    super.hasTypeBound()
```

Comment:
Looks like this line results in `ERROR: 'super' is ambiguous because Wildcard extends more than one type`, so we'll probably need `BoundedType.super.hasTypeBound()` instead. Note that this isn't really ambiguous, since the `@wildcard` type is a db-type, which can't have any members, but the QL compiler doesn't consider this special case.  I've reported this internally, so this might be fixed at some point.